language-support/ts/daml-types/index.ts

```diff
@@ -419,4 +420,82 @@ export const TextMap = <T>(t: Serializable<T>): Serializable<TextMap<T>> => ({
   }
 });
 
-// TODO(MH): `Map` type.
+/**
+ * The counterpart of DAML's `DA.Map.Map K V` type.
+ *
+ * @typeparam K The type of the map keys.
+ * @typeparam V The type of the map values.
+ */
+export interface Map<K, V> {
+  get: (k: K) => V | null;
+  getOr: (k: K, ifNotFound: V) => V;
+  has: (k: K) => boolean;
+  insert: (k: K, v: V) => Map<K, V>;
+  remove: (k: K) => Map<K, V>;
+  keys: () => K[];
+  values: () => V[];
```

Comment:
For JS maps and immutable JS those return iterators instead.

Reply:
Given that we cannot have an infinite map here, it seems to me that iterators give us a less efficient implementation in exchange for a more cumbersome API, which is not generally-speaking the kind of tradeoff I like to make.

Additionally, we happen to be targeting es5, which does not have iterators, and that seems to restrict TypeScript's ability to handle them too, so the `[...iterator]` notation to convert back to an array doesn't work (turning on `downlevelIteration` strangely didn't help here, despite what the internet says). I'm overall not too happy with how that's turned out, but I'll let you make the call.